packages/editor/src/editor/browser/data-class/variable-tree-data.ts

```diff
@@ -53,23 +53,27 @@ export const fullVariablePath = (row: Row<BrowserNode>, dontShowRootNode: boolea
 
   const parentPath = relevantParents.map(parent => parent.original.value).join('.');
 
-  if (row.original.value === 'Use entire Object' && dontShowRootNode) {
+  if (row.original.value === 'variable' && dontShowRootNode) {
     return '';
   }
   return parentPath ? `${parentPath}.${row.original.value}` : row.original.value;
 };
 
-export const rowToCreateData = (row: Row<BrowserNode>): CreateComponentData | undefined => {
+export const rowToCreateData = (
+  row: Row<BrowserNode>,
+  dontShowRootNode: boolean = false,
+  prefix?: string
+): CreateComponentData | undefined => {
   const node = row.original;
   const component = componentForType(node.info);
   if (component === undefined) {
     return undefined;
   }
-
+  const variablePath = fullVariablePath(row, dontShowRootNode);
   return {
     componentName: component.component.name,
     label: labelText(node.value),
-    value: `#{${fullVariablePath(row)}}`,
+    value: `#{${prefix ? prefix : ''}${prefix && variablePath.length > 0 ? '.' : ''}${variablePath}}`,
```

Comment:
Looks like code that should be not done in a inline thing, but in a function which is tested...